python/django/v1/ampli_app/ampli/ampli.py

```diff
@@ -0,0 +1,815 @@
+            "optional enum": optional_enum,
+            "required enum": required_enum
+        }
+
+
+class EventWithOptionalArrayTypes(BaseEvent):
+    """Event With Optional Array Types
+
+    [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Optional%20Array%20Types)
+
+    Description for event with optional array types
+
+    Owner: Test codegen
+
+    :param optional_boolean_array: Description for optional boolean array
+    :param optional_json_array: Description for optional object array
+    :param optional_number_array: Description for optional number array
+    :param optional_string_array: Description for optional string array
+    """
+
+    def __init__(
+        self,
+        optional_boolean_array: Optional[List[bool]] = None,
+        optional_json_array: Optional[List[Any]] = None,
+        optional_number_array: Optional[List[float]] = None,
+        optional_string_array: Optional[List[str]] = None
+    ):
+        super().__init__(event_type="Event With Optional Array Types")
+        self.event_properties = {
+            "optionalBooleanArray": optional_boolean_array,
+            "optionalJSONArray": optional_json_array,
+            "optionalNumberArray": optional_number_array,
+            "optionalStringArray": optional_string_array
+        }
+
+
+class EventWithOptionalProperties(BaseEvent):
+    """Event With Optional Properties
+
+    [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Optional%20Properties)
+
+    Event w optional properties description
+
+    Owner: Test codegen
+
+    :param optional_array_number: Property has no description provided in tracking plan.
+    :param optional_array_string: Property has no description provided in tracking plan.
+    :param optional_boolean: Property has no description provided in tracking plan.
+    :param optional_number: Property has no description provided in tracking plan.
+    :param optional_string: Optional String property description
+    """
+
+    def __init__(
+        self,
+        optional_array_number: Optional[List[float]] = None,
+        optional_array_string: Optional[List[str]] = None,
+        optional_boolean: Optional[bool] = None,
+        optional_number: Optional[float] = None,
+        optional_string: Optional[str] = None
+    ):
+        super().__init__(event_type="Event With Optional Properties")
+        self.event_properties = {
+            "optionalArrayNumber": optional_array_number,
+            "optionalArrayString": optional_array_string,
+            "optionalBoolean": optional_boolean,
+            "optionalNumber": optional_number,
+            "optionalString": optional_string
+        }
+
+
+class Ampli:
+    
+    def __init__(self):
+        self.client: Amplitude = None
+        self.disabled: bool = False
+
+    def load(self, options: Optional[LoadOptions] = None):
```

Comment:
Can you please add method documentation for the main methods (load, identify, group, track, etc).